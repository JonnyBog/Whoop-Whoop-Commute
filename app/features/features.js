import RootContainer from 'features/root/root-container';
import HomeContainer from 'features/home/home-container';

export {
  RootContainer as Root,
  HomeContainer as Home
};
