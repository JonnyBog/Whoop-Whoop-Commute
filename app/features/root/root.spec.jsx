import React from 'react';
import rendererShallow from 'react-test-renderer/shallow';

import { Root } from 'features';

describe('Features', () => {
  describe('Root component', () => {
    it('should export a class', () => {
      expect(Root).toBeDefined();
    });

    let props;

    afterEach(() => {
      props = null;
    });

    /**
     * createShallowRenderTree
     * @param {Object} updatedProps - defaults to props
     * @returns {JSX} shallow renderer component
     */
    function createShallowRenderTree (updatedProps = props) {
      const shallowRenderer = rendererShallow.createRenderer();

      return shallowRenderer.render(
        <Root {...updatedProps} />
      );
    }

    describe('Appearance', () => {
      it('should render correctly', () => {
        const tree = createShallowRenderTree();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
