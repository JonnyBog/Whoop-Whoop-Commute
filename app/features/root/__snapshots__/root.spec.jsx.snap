// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features Root component Appearance should render correctly 1`] = `
<React.Fragment>
  <Header />
  <Switch>
    <Switch>
      <Route
        exact={true}
        path="/"
        render={[Function]}
      />
      <Route
        path="*"
        render={[Function]}
      />
    </Switch>
  </Switch>
</React.Fragment>
`;
